lib/bundler/lockfile_parser.rb

```diff
@@ -171,43 +171,45 @@ def parse_source(line)
       end
     end
 
-    NAME_VERSION = '(?! )(.*?)(?: \(([^-]*)(?:-(.*))?\))?'.freeze
-    NAME_VERSION_2 = /^ {2}#{NAME_VERSION}(!)?$/
-    NAME_VERSION_4 = /^ {4}#{NAME_VERSION}$/
-    NAME_VERSION_6 = /^ {6}#{NAME_VERSION}$/
+    NAME_VERSION = /^( {2}| {4}| {6})(?! )(.*?)(?: \(([^-]*)(?:-(.*))?\))?(!)?$/
```

Comment:
Could I recommend this syntax:
```ruby
%r{
  ^( {2}| {4}| {6})              # Match 2,4, or 6 spaces
    (?! )                        # Match ...
    (.*?)
    (?: \(([^-]*)
    (?:-(.*))?\))?(!)?
$}x
```

The `x` at the end of that lets you add comments and split up the pieces in newlines, indent, etc. I've otherwise found regexes difficult to reason when coming from an out of context position.

Similarly, I tend to leave a comment above the regex indicating what it _should_ match. This lets someone easily match between the regex, the comments, and the example - and easily see if a change is needed.